presto-spi/src/main/java/io/prestosql/spi/connector/ColumnMetadata.java

```diff
@@ -156,4 +181,82 @@ public boolean equals(Object obj)
                 Objects.equals(this.extraInfo, other.extraInfo) &&
                 Objects.equals(this.hidden, other.hidden);
     }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+    {
+        private String name;
+        private Type type;
+        private boolean nullable = true;
+        private String comment;
+        private String extraInfo;
+        private boolean hidden;
+        private Map<String, Object> properties = emptyMap();
+
+        private Builder() {}
+
+        public Builder setName(String name)
+        {
+            this.name = name;
+            return this;
+        }
+
+        public Builder setType(Type type)
+        {
+            this.type = type;
+            return this;
+        }
+
+        public Builder setNullable(boolean nullable)
+        {
+            this.nullable = nullable;
+            return this;
+        }
+
+        public Builder setComment(@Nullable String comment)
+        {
+            this.comment = comment;
+            return this;
+        }
+
+        public Builder setComment(Optional<String> comment)
+        {
+            this.comment = requireNonNull(comment, "comment is null").orElse(null);
+            return this;
+        }
+
+        public Builder setExtraInfo(@Nullable String extraInfo)
```

Comment:
What is the point of passing nulls? In case of null I would prefer to not to call this method at all.

Reply:
it allows you to 
```
.setExtraInfo(calcualteExtraInfo(...))
```

Reply:
Let `calcualteExtraInfo` return `Optional`

Reply:
Hm.. Yes, I can make `setExtraInfo` take `Optional`